test/hotspot/jtreg/compiler/c2/irTests/scalarReplacement/AllocationMergesTests.java

```diff
@@ -0,0 +1,939 @@
+        Point p2 = new Point(x, x);
+        Point p = new Point(y, y);
+        int res = 0;
+
+        p.x = p.x * y;
+
+        if (cond)
+            p = new Point(y, y);
+
+        dummy(p2);
+
+        for (int i=3; i<324; i++)
+            res += p.x + i * x;
+
+        return res;
+    }
+
+    @Test
+    @Arguments({ Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH })
+    @IR(counts = { IRNode.ALLOC, "1" })
+    int testCmpMergeWithNull_Second(boolean cond, int x, int y) {
+        Point p = null;
+
+        if (cond)
+            p = new Point(x*x, y*y);
+
+        dummy(x);
+
+        if (p != null)
+            return p.x * p.y;
+        else
+            return 1984;
+    }
+
+    @Test
+    @Arguments({ Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH })
+    @IR(counts = { IRNode.ALLOC, "1" })
+    int testObjectIdentity(boolean cond, int x, int y) {
+        Point o = new Point(x, y);
+
+        if (cond) {
+            o = global;
+            dummy();
+        }
+
+        dummy();
+
+        return o == global ? o.x + o.y : 0;
+    }
+
+
+    @Test
+    @Arguments({ Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH })
+    @IR(counts = { IRNode.ALLOC, "2" })
+    // The merge won't be reduced because there is a SafePoint/Call referring to the merge
+    int testSubclassesTrapping(boolean c1, boolean c2, int x, int y, int w, int z) {
+        new A();
+        Root s = new Home(x, y);
+        new B();
+
+        if (c1) {
+            new C();
+            s = new Etc("Hello");
+            new D();
+        }
+        else {
+            new E();
+            s = new Usr(y, x, z);
+            new F();
+        }
+
+        dummy();
+
+        return s.a;
+    }
+
+    // ------------------ Some Objects Will be Scalar Replaced in These Tests ------------------- //
+
+
+    @Test
+    @Arguments({ Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH })
+    @IR(failOn = { IRNode.ALLOC })
+    // Since there is no Call/SafePoint after the merge we can reduce the allocation
+    int testSubclasses(boolean c1, boolean c2, int x, int y, int w, int z) {
+        new A();
+        Root s = new Home(x, y);
+        new B();
+
+        if (c1) {
+            new C();
+            s = new Etc("Hello");
+            new D();
+        }
+        else {
+            new E();
+            s = new Usr(y, x, z);
+            new F();
+        }
+
+        new G();
+
+        return s.a;
+    }
+
+    @Test
+    @Arguments({ Argument.RANDOM_EACH, Argument.RANDOM_EACH, Argument.RANDOM_EACH })
+    @IR(counts = { IRNode.ALLOC, "2" })
+    Point testNestedObjectsObject(boolean cond, int x, int y) {
+        Picture p = new Picture(x, x, y);
+
+        if (cond)
+            p = new Picture(y, y, x);
+
+        return p.position;
```

Comment:
Hi, you may want to add a test case in which this field does not escape. Also, since this is a really large test, reducing the number of warmup iterations will reduce the runtime of the test. Thanks.